src/app/atoms/Avatar.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { memo } from 'react';
+
+import clsx from 'clsx';
+import { omit } from 'lodash';
+
+import { Button, ButtonProps } from './Button';
+
+interface AvatarPropsBase {
+  size: 24 | 32 | 60;
+  elementType?: 'div' | 'button';
+  borderColor?: 'secondary' | 'gray';
+}
+
+interface AvatarDivProps extends AvatarPropsBase, React.HTMLAttributes<HTMLDivElement> {
+  elementType?: 'div';
+}
+
+interface AvatarButtonProps extends AvatarPropsBase, ButtonProps {
+  elementType: 'button';
+}
+
+export type AvatarProps = AvatarDivProps | AvatarButtonProps;
+
+const sizeButtonClassNames = {
+  24: 'w-6 h-6 rounded border p-px',
+  32: 'w-8 h-8 rounded-md border p-px',
+  60: 'w-15 h-15 rounded-2.5 border-1.5 p-[0.5px]'
+};
+
+const sizeContainerClassNames = {
+  24: 'rounded-0.75',
+  32: 'rounded',
+  60: 'rounded-lg border border-white'
+};
+
+export const Avatar = memo<PropsWithChildren<AvatarProps>>(
+  ({ children, size, className, borderColor = size === 60 ? 'gray' : 'secondary', ...restProps }) => {
```

Comment:
Sorry, it cannot be done because `elementType` is a discriminator for `restProps`, which can be either `React.HTMLAttributes<HTMLDivElement>` or `ButtonProps` without some props